src/rt/rwlock.rs

```diff
@@ -0,0 +1,269 @@
+use crate::rt::{
+    object::{self, Object},
+    thread, Access, Synchronize,
+};
+
+use std::collections::HashSet;
+use std::sync::atomic::Ordering::{Acquire, Release};
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub(crate) struct RwLock {
+    obj: Object,
+}
+
+#[derive(Debug, PartialEq)]
+enum RwLockType {
```

Comment:
Nit: i'm not a huge fan of this naming...but I'm not really sure what would be better off the top of my head. Maybe `Locked`?